### PR TITLE
Use ConfigManager for choosing which address to listen/connect to

### PR DESCRIPTION
**Pull request by Michael-Tu**

### CODE REVIEW

File: examples/run_mock_protocol_client_main.cc [hunk 1]
```diff
@@ -43,14 +54,39 @@ void LogRequestAndResponse(T request, StatusOr<U> reply_or) {
...
+  if (!config->HasMasterServer(master_name)) {
+    LOG(ERROR) << "No master server found in config: " << master_name;
+    return 1;
+  } else if (!config->HasChunkServer(chunk_server_name)) {
+    LOG(ERROR) << "No chunk server found in config: " << chunk_server_name;
+    return 1;
+  }
+
   // Initialize an instance of communication manager
   // TODO(tugan): add support to listen on host:port based on configuration
```
Comment (xicheng87) [inline examples/run_mock_protocol_client_main.cc:76]: Is this TODO comment still relevant?

File: src/server/chunk_server/run_chunk_server_main.cc [hunk 2]
```diff
@@ -1,23 +1,49 @@
...
+  if (!config->HasChunkServer(chunk_server_name)) {
+    LOG(ERROR) << "No chunk server found in config: " << chunk_server_name;
+    return 1;
+  }
+
+  LOG(INFO) << "Running as chunk server: " << chunk_server_name;
+  LOG(INFO) << "Server starting...";
   ServerBuilder builder;
 
   // TODO(tugan): add support to listen on host:port based on configuration
```
Comment (xicheng87) [inline src/server/chunk_server/run_chunk_server_main.cc:44]: Ditto.

File: src/server/master_server/run_master_server_main.cc [hunk 3]
```diff
@@ -1,26 +1,49 @@
...
+  if (!config->HasMasterServer(master_name)) {
+    LOG(ERROR) << "No master server found in config: " << master_name;
+    return 1;
+  }
+
+  LOG(INFO) << "Running as master server: " << master_name;
+  LOG(INFO) << "Server starting...";
   ServerBuilder builder;
 
   // TODO(tugan): add support to listen on host:port based on configuration
```
Comment (xicheng87) [inline src/server/master_server/run_master_server_main.cc:44]: Ditto.

File: examples/run_mock_protocol_client_main.cc (hunk 1 above)
Comment (Michael-Tu) [reply examples/run_mock_protocol_client_main.cc]: excellent catch! removed

**Pull request merged**